elastic/kibana
# [maps] add context for 'No longer contained' geo-containment alert

Fixes https://github.com/elastic/kibana/issues/134772 and https://github.com/elastic/kibana/issues/127191

PR updates geo containment executor to populate context for recovered alerts.

Recovered alert context is populated with:
* entityId
* entityDateTime
* entityDocumentId
* entityLocation
* detectionDateTime

Recovered alert context does not contain:
* containingBoundaryId
* containingBoundaryName

To test
* Run [faketracks](https://github.com/thomasneirynck/faketracks)
* Create data view for `tracks`index
* Create new map
    * add top hits per entity layer to display tracks data source.
    * fit to tracks bounds
    * add new "create index" layer called `boundaries`. Draw serval boundaries that will intersect with faketracks
* Create new alerting rule for geo_containment alert, see [tutorial](https://www.elastic.co/guide/en/kibana/current/asset-tracking-tutorial.html#_step_2_configure_an_alert) for more details
    * Set entity `INDEX` to `tracks
    * Set entity `BY` to `entity_id`
    * Set boundary `INDEX` to`boundaries`.
    * Create `Server log`  connector
    * Set `Run when` to "No longer contained"
    * Set message to:
    ```
    {
       "entityId": "{{context.entityId}}",
       "entityDateTime": "{{context.entityDateTime}}",
       "entityDocumentId": "{{context.entityDocumentId}}",
       "detectionDateTime": "{{context.detectionDateTime}}",
       "entityLocation": "{{context.entityLocation}}",
    }
    ```

Server log should display recovered alerts with context
<img width="500" alt="Screen Shot 2022-07-18 at 10 24 00 AM" src="https://user-images.githubusercontent.com/373691/179560115-d73ea844-c24b-4a55-9074-1f06daa01974.png">
